EIPS/eip-4519.md

```diff
@@ -0,0 +1,239 @@
+ 
+## Motivation
+There are nowadays many smart assets, like IoT devices, that can interact actively with other blockchain participants, that is, they can have an Ethereum account. Hence, we propose the SmartNFTs, which have the attribute "asset" to represent the address associated with the physical smart asset. Besides, a smart asset is not only a possession of an owner but also an active agent that obeys to a user to carry out certain tasks in an application. Hence, the SmartNFTs have the NFT attribute "user", which is the address of the user of the smart asset. In order to represent the secure dynamic activity of the smart asset, six more attributes are proposed: "timestamp", "timeout", "state", "hashK_OA", "hashK_UA", and "dataEngage".  The attribute "timestamp" registers in the blockchain whenever the smart asset checks it is bound to its token. The attribute "timeout" is the maximum delay time established for the asset to prove again the bond. Token "state" is included to trace the operating mode of the smart asset while "hashK_OA", "hashK_UA", and "dataEngage" are public data added to trace if secure communication channels are established between the smart asset and its owner and user. SmartNFTS extend ERC-721 NFTs, which are designed to represent passive assets, which only have an owner that owns the token, can transfer ownership, and can approve others to act in its name. 
+
+## Specification
+This kind of non-fungible token is designed to represent a secure smart asset with the capacity of generating its own account and, hence, interacting with Ethereum chain. Besides, this SmartNFT can assign the token to a user and allows the establishment of secure communication channels with the owner and user. The following attributes are proposed for the token:
+
+| Type | Name of variable | Defined by ERC-721 | Optional |
+|--|--|--|--|
+| uint256 | tokenId | Yes | No |
+| address | owner | Yes | No |
+| address | approved | Yes | Yes |
+| address | approvedForAll | Yes | Yes |
+| address | asset | No | Yes |
+| address | user | No | Yes |
+| enum | state | No | No |
+| uint256 | hashK_OD | No | Yes |
+| uint256 | HashK_UD | No | Yes |
+| uint256 | dataEngagement | No | Yes |
+| uint256 | timeStamp | No | No |
+| uint256 | timeout | No | No |
+
+The attributes do not need to be given in a specific order. The designer of the smart contract defines the order. Like in the ERC-721, the role of operator (approvedForAll) is an attribute more related to the owner than to the token.
+ 
+Asset and user are optional attributes but at least one of them should be used in a SmartNFT. In the case of using only the attribute user, two states define if the token is assigned or not to a user. Figure 1 shows the corresponding states in a flow chart. When a token is created, transferred or unassigned, the token state is set to “notAssigned”. If the token is assigned to a valid user, the state is set to "userAssigned".
```

Comment:
I think you need to define what `user` and `asset` actually _represent_. What does a token having a user mean?